### PR TITLE
TRUNK-5000: Get rid of hack that allows first component to be an integer

### PR DESCRIPTION
**Pull request by kavyakavuri**
## Description of what I changed

Removed hack from HL7ServiceImpl, getLocation now handles location_id and name directly

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5000

## Checklist: I completed these to help reviewers :)

- [X] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/LocationServiceTest.java [hunk 1]
```diff
@@ -228,7 +228,23 @@ public void getLocation_shouldReturnNullWhenNoLocationMatchGivenLocationId() {
...
 		Assert.assertNull(Context.getLocationService().getLocation("Princeton Plainsboro"));
 	}
-	
+
+	/**
+	 * @see LocationService#getLocation(String,String)
+	 */
+	@Test
+	public void getLocationWithAttribute_shouldReturnNullWhenNoLocationMatchGivenLocationId() {
+		Assert.assertNull(Context.getLocationService().getLocation("1337","locationId"));
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/api/LocationServiceTest.java]: How can you ensure that there is no location with locationId as 1337? Better to prove that before testing your implementation.

File: api/src/test/java/org/openmrs/api/LocationServiceTest.java [hunk 2]
```diff
@@ -228,7 +228,23 @@ public void getLocation_shouldReturnNullWhenNoLocationMatchGivenLocationId() {
...
+	}
+
+	/**
+	 * @see LocationService#getLocation(String,String)
+	 */
+	@Test
+	public void getLocationWithAttribute_shouldReturnNullWhenNoLocationMatchGivenLocationName() {
+		Assert.assertNull(Context.getLocationService().getLocation("Princeton Plainsboro","name"));
```
Comment (suthagar23) [inline api/src/test/java/org/openmrs/api/LocationServiceTest.java]: How can you ensure that there is no location with locationName as mentioned here? Better to prove that before testing your implementation.

File: api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java [hunk 3]
```diff
@@ -532,15 +532,10 @@ public Integer resolvePersonId(XCN xcn) throws HL7Exception {
...
 	public Integer resolveLocationId(PL pl) throws HL7Exception {
-		// TODO: Get rid of hack that allows first component to be an integer
-		// location.location_id
 		String pointOfCare = pl.getPointOfCare().getValue();
 		String facility = pl.getFacility().getUniversalID().getValue();
-		// HACK: try to treat the first component (which should be "Point of
-		// Care" as an internal openmrs location_id
 		try {
-			Integer locationId = Integer.valueOf(pointOfCare);
-			Location l = Context.getLocationService().getLocation(locationId);
```
Comment (wluyima) [inline api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java]: I don't think this is the change I expected, all you're doing is moving the logic from the service to the dao layer, I think the TODO is saying that the first component isn't supposed to be an id but doesn't suggest which component should be used instead.

**Comment by dkayiwa** [comment]
@kavyakavuri are you still working on this?

**Comment by dkayiwa** [comment]
Closing because the author no longer works on it.

**Pull request closed without merging**